Mathlib/RingTheory/Localization/Module.lean

```diff
@@ -70,83 +147,66 @@ variable [hA : IsLocalization (Algebra.algebraMapSubmonoid A S) Aₛ]
 
 open Submodule
 
+theorem IsLocalizedModule_of_isScalarTower :
+    IsLocalizedModule S (IsScalarTower.toAlgHom R A Aₛ).toLinearMap where
+  map_units s := by
+    have := IsUnit.map (Algebra.lmul R Aₛ) <| hA.map_units <|
+      ⟨algebraMap R A s, Algebra.mem_algebraMapSubmonoid_of_mem s⟩
+    rwa [← IsScalarTower.algebraMap_apply, ← AlgHom.algebraMap_eq_apply _ rfl] at this
+  surj' x := by
+    obtain ⟨⟨a, ⟨_, ⟨s, hs, rfl⟩⟩⟩, h⟩ := hA.surj' x
+    refine ⟨⟨a, ⟨s, hs⟩⟩, ?_⟩
+    rw [Submonoid.mk_smul, AlgHom.toLinearMap_apply, IsScalarTower.coe_toAlgHom', ← h,
+      ← IsScalarTower.algebraMap_apply, mul_comm, Algebra.smul_def]
+  exists_of_eq h := by
+    obtain ⟨⟨_, ⟨r, hrS, rfl⟩⟩, h⟩ := hA.exists_of_eq h
+    refine ⟨⟨r, hrS⟩, ?_⟩
+    rw [Submonoid.mk_smul, Submonoid.mk_smul, Algebra.smul_def, Algebra.smul_def, h]
+
 theorem LinearIndependent.localization_localization {ι : Type*} {v : ι → A}
-    (hv : LinearIndependent R v) : LinearIndependent Rₛ (algebraMap A Aₛ ∘ v) := by
-  rw [linearIndependent_iff'] at hv ⊢
-  intro s g hg i hi
-  choose! a g' hg' using IsLocalization.exist_integer_multiples S s g
-  have h0 : algebraMap A Aₛ (∑ i in s, g' i • v i) = 0 := by
-    apply_fun ((a : R) • ·) at hg
-    rw [smul_zero, Finset.smul_sum] at hg
-    rw [map_sum, ← hg]
-    refine' Finset.sum_congr rfl fun i hi => _
-    rw [← smul_assoc, ← hg' i hi, Algebra.smul_def, map_mul, ← IsScalarTower.algebraMap_apply, ←
-      Algebra.smul_def, algebraMap_smul, Function.comp_apply]
-  obtain ⟨⟨_, r, hrS, rfl⟩, hr : algebraMap R A r * _ = 0⟩ :=
-    (IsLocalization.map_eq_zero_iff (Algebra.algebraMapSubmonoid A S) _ _).1 h0
-  simp_rw [Finset.mul_sum, ← Algebra.smul_def, smul_smul] at hr
-  specialize hv s _ hr i hi
-  rw [← (IsLocalization.map_units Rₛ a).mul_right_eq_zero, ← Algebra.smul_def, ← hg' i hi]
-  exact (IsLocalization.map_eq_zero_iff S _ _).2 ⟨⟨r, hrS⟩, hv⟩
+    (hv : LinearIndependent R v) : LinearIndependent Rₛ ((algebraMap A Aₛ) ∘ v) := by
+  have : IsLocalizedModule S (IsScalarTower.toAlgHom R A Aₛ).toLinearMap :=
+    IsLocalizedModule_of_isScalarTower S Aₛ
+  exact hv.isLocalizedModule Rₛ S (IsScalarTower.toAlgHom R A Aₛ).toLinearMap
 #align linear_independent.localization_localization LinearIndependent.localization_localization
 
 theorem SpanEqTop.localization_localization {v : Set A} (hv : span R v = ⊤) :
     span Rₛ (algebraMap A Aₛ '' v) = ⊤ := by
```

Comment:
Since `span = ⊤` appears as both hypothesis and conclusion, I'd call this `span_eq_top_localization_localization`